libvips/create/text.c

```diff
@@ -512,29 +512,25 @@ vips_text_build(VipsObject *object)
 	g_mutex_unlock(vips_text_lock);
 
 	if (text->rgba) {
-		int y;
-
 		/* Cairo makes pre-multipled BRGA -- we must byteswap and
 		 * unpremultiply.
 		 */
-		for (y = 0; y < image->Ysize; y++)
+		for (int y = 0; y < in->Ysize; y++)
 			vips__premultiplied_bgra2rgba(
-				(guint32 *)
-					VIPS_IMAGE_ADDR(image, 0, y),
-				image->Xsize);
+				(guint32 *) VIPS_IMAGE_ADDR(in, 0, y),
+				in->Xsize);
 	}
 	else {
 		/* We just want the alpha channel.
 		 */
-		if (vips_extract_band(image, &t[1], 3, NULL) ||
-			vips_copy(t[1], &t[2],
-				"interpretation", VIPS_INTERPRETATION_MULTIBAND,
-				NULL))
+		if (vips_extract_band(in, &t[1], 3, NULL))
 			return -1;
-		image = t[2];
+		in = t[1];
+
+		in->Type = VIPS_INTERPRETATION_MULTIBAND;
```

Comment:
Is there a danger that `t[1]` could be a shared image from the cache? I think this is why we originally had a `copy()` here (the official safe way to change `Type`).

Reply:
Hmm, good point. Perhaps we should investigate other occurrences of this as well, e.g.:
https://github.com/libvips/libvips/blob/d52a983feb02c955f6422d862b27481b82e18fd6/libvips/create/point.c#L111-L113

But let's do that as follow-up. Reverted with commit ce89e78a1d2ef438c0ee9c03a4dfe2fd65fba15e.